cmd/kubectl/kubectlcobra/run.go

```diff
@@ -0,0 +1,140 @@
+package kubectlcobra
+
+import (
+	"context"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/kubectl/pkg/cmd/apply"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/kstatus/wait"
+	"time"
+)
+
+func newApplier(factory util.Factory, ioStreams genericclioptions.IOStreams) *Applier {
+	return &Applier{
+		applyOptions: apply.NewApplyOptions(ioStreams),
+		statusOptions: &StatusOptions{},
+		factory: factory,
+		ioStreams: ioStreams,
+	}
+}
+
+type Applier struct {
+	factory util.Factory
+	ioStreams genericclioptions.IOStreams
+
+	applyOptions *apply.ApplyOptions
+	statusOptions *StatusOptions
+	resolver *wait.Resolver
+}
+
+func (a *Applier) Initialize(cmd *cobra.Command) error {
+	a.applyOptions.PreProcessorFn = PrependGroupingObject(a.applyOptions)
+	err := a.applyOptions.Complete(a.factory, cmd)
+	if err != nil {
+		return err
+	}
+
+	resolver, err := a.newResolver(2 * time.Second)
+	if err != nil {
+		return err
+	}
+	a.resolver = resolver
+	return nil
+}
+
+func (a *Applier) SetFlags(cmd *cobra.Command) {
+	a.applyOptions.DeleteFlags.AddFlags(cmd)
+	a.applyOptions.RecordFlags.AddFlags(cmd)
+	a.applyOptions.PrintFlags.AddFlags(cmd)
+	a.statusOptions.AddFlags(cmd)
+	a.applyOptions.Overwrite = true
+}
+
+func (a *Applier) newResolver(pollInterval time.Duration) (*wait.Resolver, error) {
+	config, err := a.factory.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	mapper, err := a.factory.ToRESTMapper()
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := client.New(config, client.Options{Scheme: scheme.Scheme, Mapper: mapper})
+	if err != nil {
+		return nil, err
+	}
+
+	return wait.NewResolver(c, mapper, pollInterval), nil
+}
+
+func (a *Applier) run(ctx context.Context) <-chan Event {
+	ch := make(chan Event)
+
+	go func() {
+		defer close(ch)
+		adapter := &KubectlPrinterAdapter{
+			ch: ch,
+		}
+		a.applyOptions.ToPrinter = adapter.toPrinterFunc()
```

Comment:
too bad this isn't an interface

Reply:
Yeah, that would have made this a bit cleaner.